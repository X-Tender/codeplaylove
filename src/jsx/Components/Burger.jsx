import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { toggleMenu } from 'reducers/menu';

class Burger extends Component {
	constructor(props) {
		super(props);
		this.onMenuButtonClick = ::this.onMenuButtonClick;
	}

	onMenuButtonClick() {
		this.props.toggleMenu();
	}

	render() {
		const classes = `burger ${this.props.isOpen ? 'isOpen' : ''}`;
		return (
			<button className={classes} onClick={this.onMenuButtonClick} title="Menu" type="button">
				<span className="burger__line burger__line--top" />
				<span className="burger__line burger__line--center" />
				<span className="burger__line burger__line--bottom" />
			</button>
		);
	}
}

Burger.propTypes = {
	isOpen: PropTypes.bool,
	toggleMenu: PropTypes.func.isRequired,
};

Burger.defaultProps = {
	isOpen: false,
};

const mapStateToProps = ({ menu }) => ({
	isOpen: menu.isOpen,
});

const mapDispatchToProps = {
	toggleMenu,
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Burger);
