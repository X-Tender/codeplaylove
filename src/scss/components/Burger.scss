.burger {
	position: fixed;
	left: 100%;
	top: 0px;
	z-index: 1;
	border: none;
	size: 50px;
	display: flex;
	align-items: center;
	background-color: white;
	border-radius: $globalBorderRadius;
	border-top-right-radius: initial;
	border-top-left-radius: initial;
	border-bottom-left-radius: initial;
	box-shadow: 2px 0 10px rgba(#000, 0.5);
	outline: none;

	@include sm {
		top: 40px;
		border-top-right-radius: $globalBorderRadius;
	}

	&__line {
		background-color: $primaryColor;
		width: 50%;
		height: 4px;
		left: 25%;
		position: absolute;

		&--top {
			bottom: 60%;
			transition: bottom $defaultTransitionDuration ease;

			^^&.isOpen & {
				bottom: calc(50% - 2px);
			}

			^^&:not(.isOpen):hover & {
				bottom: calc(60% +2px);
			}

			^^&.isOpen:hover & {
				bottom: calc(60% -2px);
			}
		}

		&--center {
			top: calc(50% - 2px);
		}

		&--bottom {
			top: 60%;
			transition: top $defaultTransitionDuration ease;

			^^&.isOpen & {
				top: calc(50% - 2px);
			}

			^^&:not(.isOpen):hover & {
				top: calc(60% +2px);
			}

			^^&.isOpen:hover & {
				top: calc(60% -2px);
			}
		}
	}
}
