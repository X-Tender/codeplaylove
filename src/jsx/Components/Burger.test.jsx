import React from 'react';
import { shallow, mount } from 'enzyme';
import { Burger } from './Burger';

describe('Burger', () => {
	const toggleMenu = jest.fn();
	let burger = shallow(<Burger toggleMenu={toggleMenu} />);

	it('renders correctly', () => {
		expect(burger).toMatchSnapshot();
	});

	burger = shallow(<Burger isOpen toggleMenu={toggleMenu} />);
	it('renders correctly when isOpen prop is given', () => {
		expect(burger.find('button').hasClass('isOpen')).toBe(true);
	});

	describe('when mounted', () => {
		beforeEach(() => {
			const props = {
				isOpen: false,
				toggleMenu,
			};
			burger = mount(<Burger {...props} />);
		});

		it('dispatches the `toggleMenu()` method when menuButton was clicked', () => {
			burger.find('button').simulate('click');
			expect(toggleMenu).toHaveBeenCalled();
		});
	});
});
