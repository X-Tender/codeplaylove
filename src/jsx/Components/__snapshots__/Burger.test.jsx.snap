// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Burger renders correctly 1`] = `
<button
  className="burger isOpen"
  onClick={[Function]}
  title="Menu"
  type="button"
>
  <span
    className="burger__line burger__line--top"
  />
  <span
    className="burger__line burger__line--center"
  />
  <span
    className="burger__line burger__line--bottom"
  />
</button>
`;
